iceoryx_hoofs/source/error_handling/error_handling.cpp

```diff
@@ -17,43 +17,26 @@
 
 #include "iceoryx_hoofs/error_handling/error_handling.hpp"
 
-#include "iceoryx_hoofs/log/logger.hpp"
-#include "iceoryx_hoofs/log/logging.hpp"
-#include "iceoryx_hoofs/log/logmanager.hpp"
-
-#include <iostream>
-#include <sstream>
-
 namespace iox
 {
-const char* ErrorHandler::ERROR_NAMES[] = {ICEORYX_ERRORS(CREATE_ICEORYX_ERROR_STRING)};
+const char* ERROR_NAMES[] = {ICEORYX_ERRORS(CREATE_ICEORYX_ERROR_STRING)};
 
 // NOLINTNEXTLINE(cert-err58-cpp) ErrorHander only used in tests
-HandlerFunction ErrorHandler::handler = {ErrorHandler::defaultHandler};
+iox::HandlerFunction ErrorHandler::handler = {ErrorHandler::defaultHandler};
 
-std::mutex ErrorHandler::handler_mutex;
-
-std::ostream& operator<<(std::ostream& stream, Error value) noexcept
+const char* toString(const Error error) noexcept
 {
-    stream << ErrorHandler::toString(value);
-    return stream;
+    return ERROR_NAMES[static_cast<typename std::underlying_type<Error>::type>(error)];
 }
 
-void ErrorHandler::defaultHandler(const Error error,
-                                  const std::function<void()>& errorCallBack,
+void ErrorHandler::defaultHandler(const uint32_t error IOX_MAYBE_UNUSED,
```

Comment:
Actually the C++ best practice is here just to remove the variable name and write it like: `const uint32_t, const char * errorName`.
Then it is clear we ignore the argument and we do not have to use `IOX_MAYBE_UNUSED` to suppress compiler warnings.